servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -6402,6 +6403,147 @@ void enforceGroupUserAuthorityRestrictions(final String domainName, final String
         }
     }
 
+    /**
+     * This method returns list of Principals based on the state parameter supplied
+     * @param queriedState state of principal
+     * @return List of Principals from DB
+     */
+    List<Principal> getPrincipals(int queriedState) {
+        List<Principal> principals = new ArrayList<>();
+        Principal principal;
+        try (ObjectStoreConnection con = store.getConnection(true, false)) {
+           List<String> dbPrincipals = con.getPrincipals(queriedState);
+            Principal.State principalState = Principal.State.getState(queriedState);
+           for (String dbPrincipal : dbPrincipals) {
+               principal = ZMSUtils.createPrincipalForName(dbPrincipal, zmsConfig.getUserDomain(), null);
+               ((SimplePrincipal) principal).setState(principalState);
+               principals.add(principal);
+           }
+        }
+        return principals;
+    }
+
+    /**
+     * This method toggles state for supplied Principals based on the flag in DB
+     * as well as modifies memberships of all roles and groups of current principal(s)
+     * @param changedPrincipals List of Principals from User Authority
+     * @param suspended boolean indicating principal's state
+     */
+    void updatePrincipalByStateFromAuthority(List<Principal> changedPrincipals, boolean suspended) {
+        final String caller = "updatePrincipalByStateFromAuthority";
+        List<Principal> updatedUsers = new ArrayList<>();
```

Comment:
this should be within the if block so we're not wasting create the listing if there is nothing to do. or better yet in order not to have the whole function as we have now within an if block, it's simpler just do the check up front and return and that makes it easier to follow:

if (chargedPrincipals.isEmpty()) {
  return;
}

// continue with the rest of the method

final String caller = "updatePrincipalByStateFromAuthority";
List<Principal> updatedUsers = new ArrayList<>();
...